common/src/main/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoader.java

```diff
@@ -306,16 +308,16 @@ private static AlgorithmParameterSpec getLegacySpecForKeyStoreKey(@NonNull final
      * @return a {@link AlgorithmParameterSpec} for the keystore key (that we'll use to wrap the secret key).
      */
     private static AlgorithmParameterSpec getSpecForKeyStoreKey(@NonNull final Context context, @NonNull final String alias) {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
-            return getLegacySpecForKeyStoreKey(context, alias);
-        } else {
-            int purposes = KeyProperties.PURPOSE_WRAP_KEY  | KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT;
+        if (CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_NEW_KEY_GEN_SPEC_FOR_WRAP) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
If the issue only happens for certain devices, how do you plan to selectively turn ON and OFF the flight for specific devices.
One way would be to add legacy logic as fallback in case the new logic fails.